drivers/led/sct2024.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2025 Hubert Miś
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sct_sct2024
+
+/**
+ * @file
+ * @brief LED driver for the SCT2024 LED driver
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(sct2024, CONFIG_LED_LOG_LEVEL);
+
+#define SCT2024_LED_COUNT 16
+/* SCT2024 can be chained up to control more LEDs.
+ * The driver can be extended to support chaining.
+ */
+#define SCT2024_MAX_CHAIN_LENGTH 1
+
+struct sct2024_cfg {
+	const struct spi_dt_spec spi;
+	const struct gpio_dt_spec la_pin;
+	const struct gpio_dt_spec oe_pin;
+	const struct led_info *leds_info;
+};
+
+struct sct2024_data {
+	uint16_t led_bitmap[SCT2024_MAX_CHAIN_LENGTH];
+};
+
+static const struct led_info *sct2024_get_led_info(const struct device *dev, uint32_t led)
+{
+	const struct sct2024_cfg *cfg = dev->config;
+
+	if (led >= SCT2024_LED_COUNT * SCT2024_MAX_CHAIN_LENGTH) {
+		return NULL;
+	}
+
+	return &cfg->leds_info[led];
+}
+
+static int sct2024_get_led_index(const struct device *dev, uint32_t led)
+{
+	const struct led_info *led_info = sct2024_get_led_info(dev, led);
+
+	if (led_info == NULL) {
+		return -EINVAL;
+	}
+
+	return led_info->index;
+}
+
+static int sct2024_spi_write(const struct device *dev)
+{
+	const struct sct2024_cfg *cfg = dev->config;
+	struct sct2024_data *data = dev->data;
+	uint8_t buffer[SCT2024_MAX_CHAIN_LENGTH * 2] = {0};
+	int ret;
+
+	/* Feature proposal: When SCT2024_MAX_CHAIN_LENGTH is greater than 1, limit the number of
+	 * iterations based on the greatest led index for the chain - detect the actual chain length
+	 */
+	BUILD_ASSERT(SCT2024_MAX_CHAIN_LENGTH == 1,
+		     "Driver currently supports only a single SCT2024 device in the chain");
```

Comment:
these are evaluated at build time, bring it out of the function code, typically you'd put them either at the top or bottom of the file